### ISSUE TITLE
[BUG] zend_mm_heap on Build-in dev PHP web server

### ISSUE THREAD

**Issue by giny21** [barbushin/php-imap#541]
**Environment (please complete the following information):**
 - PHP IMAP version: 4.1
 - PHP Version: 7.4.9
 - Type of execution: Build-in dev PHP Web Server

**Describe the bug**
Ok, this is my second post on github ever so I will try to be clear as possible and get to the point quickly.
I want to check if credential send by user are correct so I create an object of class Mailbox and use
checkMailbox() method to check if connection to inbox is possible. 
At first I have got zend_mm_heap corrupted error which crashes build-in php server. After a lot of digging and some segfault errors too, I found class Imap and method open(). Interesting part of this method looks like this atm:
```php
        $result = \imap_open($mailbox, $username, $password, $options, $n_retries, $params);

        if (!$result) {
            $lastError = \imap_last_error();

            if ('' !== \trim($lastError)) {
                throw new UnexpectedValueException('IMAP error:'.$lastError);
            }

            throw new UnexpectedValueException('Could not open mailbox!', 0, self::HandleErrors(\imap_errors(), 'imap_open'));
        }
```
But there is a problem - imap_open throw ErrorException (this behavior isn't described in official documentation) which, in the set of events, cause segfault on my configuration. If we change this few line to:
```php


        try{
            $result = \imap_open($mailbox, $username, $password, $options, $n_retries, $params);

            if (!$result) {
                $lastError = \imap_last_error();
                if ('' !== \trim($lastError)) {
                    throw new UnexpectedValueException('IMAP error:'.$lastError);
                }
    
                throw new UnexpectedValueException('Could not open mailbox!', 0, self::HandleErrors(\imap_errors(), 'imap_open'));
            }
        }
        catch(ErrorException $e){
            throw new UnexpectedValueException('Could not open mailbox!', 0, self::HandleErrors(\imap_errors(), 'imap_open'));
        }
```
its starts to work and everything is alright again!
My proposition is to put this code in next pull request. I dont know how to do this properly so I just left this here, hope it will help someone in future.
**To Reproduce**
Steps to reproduce the behavior.

The used code:
```php
            $connectionArray = json_decode($request->getContent(), true);
            $inbox = $this->imapService->getInbox($connectionArray);
            $inbox->checkMailbox();
```

The headers of the parsed email, if required and possible (only, if it's NOT confidential):
```
Received: from BN3NAM04HT142.eop-NAM04.prod.protection.outlook.com
(2603:10a6:209:2a::30) by AM6PR05MB6294.eurprd05.prod.outlook.com with HTTPS
via AM6PR07CA0017.EURPRD07.PROD.OUTLOOK.COM; Sun, 5 May 2019 12:29:42 +0000
Received: from BN3NAM04FT054.eop-NAM04.prod.protection.outlook.com
(10.152.92.54) by BN3NAM04HT142.eop-NAM04.prod.protection.outlook.com
(10.152.92.244) with Microsoft SMTP Server (version=TLS1_2,
cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384) id 15.20.1835.13; Sun, 5 May
2019 12:29:41 +0000
...
```

**Expected behavior**

**Screenshots / Outputs**
If applicable, add screenshots or outputs of your script to help explain your problem.

**Additional context**

**Comment by bapcltd-marv** [comment]
may have already been addressed in the dev branch by #523 & #524

**Comment by Sebi94nbg** [comment]
Closing this as it's already very old.

Please re-open this issue, if the issue should still exist in the release [4.3.0](https://github.com/barbushin/php-imap/releases/tag/4.3.0) and newer.
